docs/DeveloperGuide.md

```diff
@@ -1,38 +1,361 @@
+
+1. The `Main` class which initialises all the other components at startup, and connects them with each other
+2. The `UI` component which is responsible for all user input and output
+3. The `Parser` component which parses user input and creates the relevant Command objects
+4. The `Command` component which executes the logic
+5. The `Accounts` component which manages the user's accounts
+6. The `Forex` (Foreign Exchange) component which handles exchange-rate related logic
+7. The `Transactions` component which manages the user's transactions
+8. The `Storage` component which handles the saving and loading of data to disk
+
+### General Sequence
+
+The following is a high-level sequence of a single `create-account SGD` command, which demonstrates how the components
+interact with each other:
+
+![BasicSequence](images/BasicSequence.png)
+
+### UI component
+
+The API of this component is specified in the `Ui.java`.
+
+![UI Class Diagram](images/UiClassDiagram.png)
+
+The UI class deals with the user interaction with the application, which includes the printing and the logic to read in
+inputs. We will pass in the instance of UI into the `execute` method of the `Command` class. The main features includes:
+
+- Printing text such as new line, spacer, farewell message, greeting message.
+
+- Empowers the users to print customised message in `printf` and `printMessage` method.
+
+- Reads in the user input as `String`.
+
+### Parser component
+
+![Parser Class Diagram](images/ParserClassDiagram.png)
+
+The `Parser` Component
+
+- Parses the user input and creates the relevant `Command` object
+- Makes use of the `CommandType` enum to determine the type of command to create
+
+### Accounts Component
+
+Here is a class diagram of the `Accounts` component
+
+![AccountListClassDiagram](images/AccountListClassDiagram.png)
+
+The `Accounts` Component
+
+- Stores the `AccountList` which contains all the user's accounts
+- `AccountList` handles all logic dealing with accounts
+- `Account` stores both its currency type and its balance
+- There can be only one `Currency` per `Account`
+- There can be only one `Account` of each `Currency`
+
+### Forex component
+
+Here is a class diagram of the `Forex` component
+![ForexClassDiagram](images/ForexClassDiagram.png)
+
+The `Forex` Component
+
+- Stores the exchange rates of 1 SGD to all supported currencies in a hash map
+- Each `Forex` object represents the relationship between two currencies
+- `convert` can be called on a `Forex` object to convert an amount using the relationship
+- Each `Forex` object has an initial and target `Currency`
+- There is only one instance of the `exchangeRates` hash map.
+
+The `Currency` Enum
+
+- Keeps all currency types supported by the exchange
+- Exchange rates are manually pulled from https://www.xe.com/currencyconverter/convert
+- Each `Forex` instance must have two `Currency` associated with it
+
+### Open Exchange Rates API
```

Comment:
Another minor comment, consider elaborating on how one can set up the program to interact with the API, such as specifying where and how to specify API tokens.